tests/diagnostics/attribute-error.slang

```diff
@@ -0,0 +1,34 @@
+// attribute.slang
+
+// Tests reflection of user defined attributes.
+
+//TEST:REFLECTION:-stage compute -entry main -target hlsl
+
+[AttributeUsage(AttributeTargets.Struct)]
```

Comment:
From a design standpoint, I'm still not convinced that aping C# exactly here is the best possible design, as compared to something like:

```
[AttributeTarget(StructDecl)]
...
```

I'm fine with the current design for now, but I think we need to "bikeshed" this part of the design before we consider this a feature we are ready for general users (people who aren't also Slang contributors) to rely on.

I *might* even argue that we should prefix the relevant names here with underscores to mark their not-ready-for-general-use status (`[_AttributeUsage(_AttributeTargets.Struct)]`).

Reply:
I tried to not diverge from C# when there is no obvious reason to.

Reply:
That's fair, but we need some way to be clear to users about what is intended to be public/stable language and library features, and what is intended to be an non-public or unstable feature that carries a "use at your own risk" warning.

I'm not ready to consider this stuff a public/stable feature just yet. I'd want us to do some serious thinking about the long-term goals for the feature before we finalize the syntax it uses today.

So long as we don't advertise this to users it is probably okay, but I just want to be clear that there is a precedent in the current design for marking "use at your own risk" features with leading underscores.